app/src/main/java/com/github/gotify/api/CertUtils.java

```diff
@@ -69,7 +74,43 @@ public static void applySslSettings(OkHttpClient.Builder builder, SSLSettings se
                             context.getSocketFactory(), (X509TrustManager) trustManagers[0]);
                 }
             }
-        } catch (Exception e) {
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                if (settings.clientCert != null) {
```

Comment:
I couldn't get client certificates working together with a self-signed server certificate, could you test if this works correctly. Maybe the SSLContext gets overridden because
```
SSLContext context = SSLContext.getInstance("TLS");
context.init(...);
```
is called two times.

Reply:
Yes, the problem was I was not adding the client cert to the key store for branch related to self-signed certs at all. It now tries to add the client cert to that as well if possible. I've retested against a real cert and it's still working. I am having issues configuring my docker gotify for a self-signed cert (I currently have an installation that uses caddy and real certs). If there's anyway you can retest with your setup in the meantime, it would be appreciated.